service/geopmdpy_test/TestPlatformService.py

```diff
@@ -96,21 +99,525 @@ def test_get_group_access_default(self):
         self.assertEqual(set(signals_expect), set(signals))
         self.assertEqual(set(controls_expect), set(controls))
 
+    def test_get_group_access_empty(self):
+        signals, controls = self._platform_service.get_group_access('')
+        self.assertEqual([], signals)
+        self.assertEqual([], controls)
+
+    def test_get_group_access_invalid(self):
+        err_msg = 'Linux group name cannot begin with a digit'
+        with self.assertRaisesRegex(RuntimeError, err_msg):
+            self._platform_service.get_group_access('1bad')
+        err_msg = 'Linux group is not defined'
+        with self.assertRaisesRegex(RuntimeError, err_msg):
+            self._platform_service.get_group_access('bad')
+
+    def test_get_group_access_default(self):
+        signals_expect = ['geopm', 'signals', 'default', 'energy']
+        controls_expect = ['geopm', 'controls', 'default', 'power']
+        self._write_group_files_helper('', signals_expect, controls_expect)
+        signals, controls = self._platform_service.get_group_access('')
+        self.assertEqual(set(signals_expect), set(signals))
+        self.assertEqual(set(controls_expect), set(controls))
+
     def test_get_group_access_named(self):
         controls_expect = ['geopm', 'controls', 'named', 'power']
-        named_dir = os.path.join(self._CONFIG_PATH.name, 'named')
-        os.makedirs(named_dir)
-        control_file = os.path.join(named_dir, 'allowed_controls')
-        lines = list(controls_expect)
+        self._write_group_files_helper('named', [], controls_expect)
+        signals, controls = self._platform_service.get_group_access('named')
+        self.assertEqual([], signals)
+        self.assertEqual(set(controls_expect), set(controls))
+
+    def test_set_group_access_invalid(self):
+        '''
+        Test that errors are encountered when there are no
+        allowed signals or controls.
+        '''
+        # Test no pio signals
+        signal = ['power']
+        err_msg = 'The service does not support any signals that match: "{}"'.format(signal[0])
+        with mock.patch('geopmdpy.pio.signal_names', return_value=[]), \
+             self.assertRaisesRegex(RuntimeError, err_msg):
+            self._platform_service.set_group_access('', signal, [])
+
+        # Test no pio signals or controls
+        control = ['power']
+        err_msg = 'The service does not support any controls that match: "{}"'.format(control[0])
+        with mock.patch('geopmdpy.pio.signal_names', return_value=[]), \
+             mock.patch('geopmdpy.pio.control_names', return_value=[]), \
+             self.assertRaisesRegex(RuntimeError, err_msg):
+            self._platform_service.set_group_access('', [], control)
+
+    def _set_group_access_test_helper(self, group):
+        '''
+        Test that the files will be created with the proper data.
+        '''
+        all_signals = ['power', 'energy']
```

Comment:
There are some strings used in the helper methods that are known by the test methods.  Might be nice to put these in members of the test class rather than have several copies of the list in the file.

Reply:
Let's talk about this one because there's a little more nuance going on with my helpers.